unit/util/lazy.cpp

```diff
@@ -16,8 +16,8 @@ SCENARIO("lazy test", "[core][util][lazy]")
     ++call_counter;
     return s.length();
   };
-  auto lazy_length =
-    lazyt<int>::from_fun([&]() { return length_with_counter("foo"); });
+  lazyt<std::size_t> lazy_length =
+    lazy([&] { return length_with_counter("foo"); });
```

Comment:
Why re-purpose a unit test?

Reply:
no good reason, I've now added this one as a new scenario instead.